File: solutions/julia/prime-factors/prime-factors.jl
function prime_factors(n)
    i = 2
    factors = Vector{Int}()

    while n > 1
        if n % i == 0
            push!(factors, i)
            n = n ÷ i
        else
            isodd(i) ? i += 2 : i += 1
        end
    end
    return factors
end

File: solutions/fsharp/prime-factors/PrimeFactors.fs
module PrimeFactors

let factors (number: int64) : int list =

    let rec nextFactor remainder candidate acc =
        match candidate with
        | _ when remainder = 1L -> acc
        | n when remainder % n = 0L -> nextFactor (remainder / n) n (acc @ [ int n ])
        | _ -> nextFactor remainder (candidate + 1L) acc

    nextFactor number 2L List.empty